src/changelogs/parse_changelogs.py

```diff
@@ -36,56 +34,166 @@ def run(self, version, logs):
+        map = {
+            'Hero Gameplay': 'Heroes',
+            'Hero Gamepla': 'Heroes',
+            'Hero': 'Heroes',
+            'Item Gameplay': 'Items',
+            'New Items': 'Items',
+            'Misc Gameplay': self.default_tag,
+            'General': self.default_tag,
+            'General Change': self.default_tag,
+        }
 
-                    changelog_dict[resource_type][resource_name].append(line)
+        # headings in this list are not converted to tags
+        headings_to_ignore = ['Ranked Mode']
+        if heading in headings_to_ignore:
+            return None
 
-            if not resource_found:
-                changelog_dict[current_heading]['General'].append(line)
+        return map.get(heading, heading)
 
-        changelog_with_icons = self._embed_icons(changelog_dict)
-        os.makedirs(self.OUTPUT_CHANGELOGS, exist_ok=True)
-        json_utils.write(self.OUTPUT_CHANGELOGS + f'/date/{version}.json', changelog_with_icons)
-        return changelog_with_icons
+    def _register_tag(self, tags, tag, is_group_tag=True):
+        """
+        Registers a tag to the changelog's current unique tags,
+        and to the static unique list of tags.
+        """
+
+        if tag not in tags:
+            tags.append(tag)
+
+        if tag not in self.unique_tags:
+            self.unique_tags.append(tag)
+
+        if is_group_tag and tag not in self.unique_tag_groups:
+            self.unique_tag_groups.append(tag)
+
+        return tags
 
-    # mass find and replace of any resource names with the ability icon template
+    # mass find and replace of referenced tags to {{PageRef|tag}}
     def _embed_icons(self, changelog):
-        for header, log_groups in changelog.items():
-            for group_name, logs in log_groups.items():
-                for index, log in enumerate(logs):
-                    for resource_key in self.resources:
-                        resource = self.resources[resource_key]
-                        resource_name = resource['Name']
-                        if resource_name is None:
-                            continue
-
-                        if resource_name not in log:
-                            continue
-                        icon = f'{{AbilityIcon|Ability={resource_name}|Size=20}}'
-                        new_log = log.replace(resource_name, f'{icon} {resource_name}')
-                        changelog[header][group_name][index] = new_log
+        new_changelog = changelog.copy()
+        for index, log in enumerate(changelog):
+            tags = log['Tags']
+            description = log['Description']
+            remaining_description = description
+            for tag in tags:
+                template = 'PageRef'
+
+                icon = '{{' + template + '|' + tag + '}}'
+
+                if tag in remaining_description:
+                    description = description.replace(tag, icon)
+                    new_changelog[index]['Description'] = description
```

Comment:
will the final description formatting be handled in another PR? Eg. arrows, bolding etc.

Reply:
By arrows, do you mean like "Abrams health 600 --> 500"? I was planning on leaving them as is, but I suppose we could create something like {{Arrow|Right}} that displays a pretty arrow instead? 

If the page name appears in the changelog, itll automatically be bolded. Though, you just reminded me that I still need to make the headings appear in a wiki friendly format.

What other formatting do you have in mind?

Reply:
The arrow thing was to match the wiki, but looking again, looks like they just use "to" now. Similar thing on other wikis, so maybe we don't need that 🤔

So maybe it doesn't actually need anything more, but I thought I'd open up the convo on it, as given the potential difficulty with deploying, we should make certain its as good as possible

Reply:
Agreed. I'm sure that as I take a light skim through merging with existing changelogs, I'll find more tags that can be created before doing the actual merge. Though, that will still be many PR's down the line, as after this we need to hook it  up to the manifest-id-versioning branch.